nixos/modules/services/misc/bazel-remote.nix

```diff
@@ -0,0 +1,146 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.bazelRemote;
+  condOption = name: value: if value != null then " --${name} ${toString value}" else "";
+  bazelRemoteConfig
+    = condOption "config_file" cfg.yamlConfig
+    + condOption "dir" cfg.dir
+    + condOption "max_size" cfg.maxSize
+    + condOption "host" cfg.bind
+    + condOption "port" cfg.port
+    + condOption "htpasswd_file" cfg.htpasswdFile
+    + condOption "tls_cert_file" cfg.tlsCertFile
+    + condOption "tls_key_file" cfg.tlsKeyFile
+    + condOption "idle_timeout" cfg.idleTimeout
+    ;
+in
+{
+
+  ###### interface
+
+  options = {
+
+    services.bazelRemote = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable the Bazel Remote server.";
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.bazel-remote;
+        defaultText = "pkgs.bazelRemote";
+        description = "Which Bazel Remote derivation to use.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "bazel";
+        description = "User account under which Bazel Remote runs.";
+      };
+
+      pidFile = mkOption {
+        type = types.path;
+        default = "/var/lib/bazelRemote/bazelRemote.pid";
+        description = "";
+      };
+
+      port = mkOption {
+        type = types.int;
+        default = 8080;
+        description = "The port for Bazel Remote to listen to.";
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to open ports in the firewall for the server.
+        '';
+      };
+
+      bind = mkOption {
+        type = with types; nullOr str;
+        default = null; # 127.0.0.1
+        description = "The IP interface to bind to.";
+        example = "127.0.0.1";
+      };
+
+      yamlConfig = mkOption {
+        type = with types; nullOr lines;
+        default = null;
+        description = "Path to a YAML configuration file. If this option is specified then all other bazel-remote options are ignored";
+      };
+
+      dir = mkOption {
+        type = types.path;
+        description = "Directory path where to store the cache contents";
```

Comment:
Does it make sense to have a default here?

Reply:
bazel-remote doesn't, I copied all the options and descriptions from the `bazel-remote --help` and `dir` and `maxSize` are required their so I'm leaving them as required here.

Reply:
Alright that's fine then.